src/core/src/validation_util.cpp

```diff
@@ -1275,7 +1275,11 @@ HostTensorPtr evaluate_bound(const Output<Node>& output, bool is_upper, bool inv
                     if (should_invalidate && input.get_target_inputs().size() == 1)
                         tensor.invalidate_values();
                 }
-                propagate_rt_info(node, output);
+                try {
+                    propagate_rt_info(node, output);
```

Comment:
[NotBlocking] Is possible to change `propagate_rt_info` to return bool value if everything is ok? Or maybe at least we can throw our own exception if sth goes wrong (instead of flow based on `bad_weak_ptr`)?

Reply:
As we discussed offline, I've moved back error handling to the propagate_rt_info.